src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -2,294 +2,450 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Collections.Generic;
 using Nethermind.Consensus.Messages;
 using Nethermind.Core;
-using Nethermind.Core.Crypto;
-using Nethermind.Core.Extensions;
 using Nethermind.Core.Specs;
+using Nethermind.TxPool;
+using Nethermind.Core.Crypto;
 using Nethermind.Crypto;
 using Nethermind.Evm;
 using Nethermind.Int256;
-using Nethermind.TxPool;
 
-namespace Nethermind.Consensus.Validators
+namespace Nethermind.Consensus.Validators;
+
+public sealed class TxValidator(ulong chainId) : ITxValidator
 {
-    public class TxValidator : ITxValidator
+    private readonly Dictionary<TxType, ITxValidator> _validators = new()
     {
-        private readonly ulong _chainIdValue;
-
-        public TxValidator(ulong chainId)
         {
-            _chainIdValue = chainId;
-        }
+            TxType.Legacy, new AllTxValidator([
+                IntrinsicGasTxValidator.Instance,
+                new LegacySignatureTxValidator(chainId),
+                ContractSizeTxValidator.Instance,
+                NonBlobFieldsTxValidator.Instance,
+            ])
+        },
+        {
+            TxType.AccessList, new AllTxValidator([
+                new ReleaseSpecTxValidator(spec => spec.IsEip2930Enabled),
+                IntrinsicGasTxValidator.Instance,
+                SignatureTxValidator.Instance,
+                new ExpectedChainIdTxValidator(chainId),
+                ContractSizeTxValidator.Instance,
+                NonBlobFieldsTxValidator.Instance,
+            ])
+        },
+        {
+            TxType.EIP1559, new AllTxValidator([
+                new ReleaseSpecTxValidator(spec => spec.IsEip1559Enabled),
+                IntrinsicGasTxValidator.Instance,
+                SignatureTxValidator.Instance,
+                new ExpectedChainIdTxValidator(chainId),
+                GasFieldsTxValidator.Instance,
+                ContractSizeTxValidator.Instance,
+                NonBlobFieldsTxValidator.Instance,
+            ])
+        },
+        {
+            TxType.Blob, new AllTxValidator([
+                new ReleaseSpecTxValidator(spec => spec.IsEip4844Enabled),
+                IntrinsicGasTxValidator.Instance,
+                SignatureTxValidator.Instance,
+                new ExpectedChainIdTxValidator(chainId),
+                GasFieldsTxValidator.Instance,
+                ContractSizeTxValidator.Instance,
+                BlobFieldsTxValidator.Instance,
+                MempoolBlobTxValidator.Instance
+            ])
+        },
+    };
+
+    public TxValidator AddValidator(TxType type, ITxValidator validator)
```

Comment:
```suggestion
    public TxValidator AddOrReplaceValidator(TxType type, ITxValidator validator)
```

Reply:
SetValidator?

Reply:
I initially went with `WithValidator` so the call site would look like:

```c#
_api.TxValidator
    .WithValidator(TxType.DepositTx, new OptimismTxValidator())
    .WithValidator(...)
```

As long as we're consistent with other classes with similar behavior any name is fine.

Reply:
Lets go with either `WithValidator` or `SetValidator`